app/src/main/java/com/wafflestudio/snutt2/components/compose/ProgressDialog.kt

```diff
@@ -74,7 +76,7 @@ fun CustomDialog(
         ) {
             Column(
                 modifier = Modifier
-                    .width(width ?: (screenWidthInDp - 50.dp))
+                    .width(dialogWidth)
```

Comment:
기존 정책)
- CustomDialog는 width의 암묵적인 기본값을 screenWidth - 50dp 로 한다. 특정 너비로 지정하고 싶으면 값을 넣어 지정한다.

이슈)
- 가로가 넓은 기기에서는 너비 screenWidth - 50dp에 대한 높이가 기기 화면 높이보다 커서, 상하 UI가 잘린다.
- 이와 별개로, 가로로 긴 기기에서는 너비 screenWidth - 50dp가 너무 넓다.

해결)
- 다이얼로그의 `적절한 기본값`이라 생각했던 screenWidth - 50dp는 가로로 긴 기기에서 적합하지 않다.
- 다이얼로그의 적절한 최대 너비는 400dp 이다. 따라서 다이얼로그 width의 암묵적인 기본값을 min(screenWidth - 50dp, 400dp) 로 수정한다

내 사고의 흐름은 이런데, 다른 거 다 필요 없이 `CustomDialog`에서 여기만
```suggestion
                    .width(width ?: min(screenWidthInDp - 50.dp, 400.dp))
```
이렇게 바꾸면 되지 않을까

Reply:
showColorPickerDialog 에 `width = 400.dp,` 을 넣었지만 내부적으로 
```
min(width ?: Int.MAX_VALUE.dp, screenWidthInDp - 50.dp)
```
라는 로직이 작용해서 일반적인 기기에서는 400과 다른 값이 들어가게 되는데, 이런 게 쌓이면 스파게티 코드가 되어 갈 것 같음


width의 값을 지정했다? -> 뭔가 디자인적인 의도가 있는 거니 무조건 그 너비로 고정!
평범한 다이얼로그다? -> dialog 너비는 쓰는 쪽이 관심 갖지 않는다. CustomDialog는 외부의 의도가 없으면(파라미터가 null이면) 기본값으로 설정한다. 그 기본값은 min(screenWidthInDp - 50.dp, 400.dp) 이다! << 우리의 정책

요런 느낌

Reply:
음 근데 이거는 나도 좀 궁금하긴 했어
사실 아직 많이 안봐서 화면 넓이를 어떻게 가져오고
그게 어떻게 400이랑 비교돼서 들어가는지 정리가 안되긴 했는데
ColorPicker에서는 따로 customdialog를 쓰는 부분이 없길래
이거 되는거 맞나? 싶기도 하고... 아직 이해가 부족하다 ㅠ

Reply:
미안 이건 내가 코드를 개같이 짜서 그래

showColorPickerDialog 보면 `modalState.setOkCance` 가 있고 마지막에 `.show()` 를 하는데, modalState는 [여기](https://github.com/wafflestudio/snutt-android/blob/develop/app/src/main/java/com/wafflestudio/snutt2/views/RootActivity.kt#L194-L195)가 출처라서 ShowModal 들어가보면 CustomDialog 있음

Reply:
아하 내가 생각한 400dp는 컬러피커의 적당한 width였어
다른 다이얼로그의 적당한 width까지 400dp라고 생각하진 않았어

- 일반적인 다이얼로그의 width -> 기존대로 screenWidth - 50dp
- 컬러피커의 width -> min(400dp, screenWidth - 50dp)

를 의도했어

Reply:
그냥 [2d329bf](https://github.com/wafflestudio/snutt-android/pull/282/commits/2d329bf66e70b021364a8a1928d25451a33fff53)처럼 showColorPicker에서 `width = min(400dp, screenWidth - 50dp)`로 하는 게 나을까?

저 커밋대로라면 `screenWidth - 50dp`는 CustomDialog의 로직인데 showCustomDialog에 있는거니까, showCustomDialog만 봤을 때 엥 싶을 것 같아서 바꾼거였는데

 형 말대로 CustomDialog가 width 파라미터와 다른 값으로 그리는 게 오히려 비직관적이겠다

Reply:
아 아니면 폴드에서 기존 다이얼로그까지 좀 이쁘게 보여줄 겸 형 말대로 `암묵적인 기본값을 min(screenWidth - 50dp, 400dp) 로 수정`할까??

Reply:
맞아 나는 앱 전반적으로 적용하는 정책으로 다이얼로그의 최대 너비를 400dp로 지정하면 좋겠다고 생각했음

컬러피커에서 UI 깨지는 문제가 터진 근본적인 원인은 컬러피커 다이얼로그의 종횡비가 1보다 커서가 아니라, 그냥 단순히 screenWidth - 50 이라는 수치가 screenWidth가 아주 큰 기기를 고려하지 않았기 때문이라고 생각했어

Reply:
다이얼로그의 전체 정책은 그대로 두고 컬러피커만 `width -> min(400dp, screenWidth - 50dp)` 라고 적용하고자 했다면 CustomDialog를 건드리지 말고 showColorPickerDialog에서 
```
modalState.set(
        // ...
        width = min(400.dp, screenWidthInDp - 50.dp),
)
```
딱 요것만 바꿔야 되지

Reply:
> 저 커밋대로라면 screenWidth - 50dp는 CustomDialog의 로직인데 showCustomDialog에 있는거니까, showCustomDialog만 봤을 때 엥 싶을 것 같아서 바꾼거였는데

요 말에서 showCustomDialog 가 아니라 showColorPickerDialog 를 말한 거지?

modalState.set에서 width의 값을 지정하기로 했다면, `충분히 넓은 기기에서는 400dp, 좁은 기기에서는 screenWidth - 50dp` 까지가 showColorPickerDialog의 정책이고 걔만의 로직이기 때문에 상관 없다고 생각함

Reply:
근데 이제 위에서 말했듯 지금 사실 CustomDialog의 공통 정책에 문제가 있는 게 근본적 원인이므로 거기를 바꿔야 하고, 마침 제안한 400dp가 예쁘므로 그걸루 하자 라는 의견 ㅋㅋㅋ

그렇게 하면 컬러피커만의 독자적인 width 정책을 적용할 필요도 없어지겠지

Reply:
👏 👏 👏  수정했습니당